src/main/webapp/dashboard.test.js

```diff
@@ -0,0 +1,132 @@
+  document.body.innerHTML = `
+  <div id="about-me" class="container tab-pane active"><br>
+    <div class="card card-holder col-12 ">
+      <div id="about-me-card-body" class="card-body">
+        <img id="user-image" class="card-img-top img-thumbnail p-2 m-2 
+            mx-auto d-block user-img" src="#" alt="Card image">
+        <h4 id="user-name-location" class="card-title dark-emph"></h4>
+        <p id="user-bio" class="card-text"></p>
+        <a id="user-github" href="#" class="btn btn-primary btn-emphasis" 
+            role="button">
+          Github
+        </a>
+        <a id="user-email" href="#" 
+            class="mailtoui btn btn-primary btn-emphasis" role="button">
+          Email Me
+        </a>
+      </div>
+    </div>
+  </div>
+  `;
+};
+
+describe('About me card of a user with no missing optional fields', () => {
+  beforeAll(() => {
+    resetDocumentBody();
+    dashboard.createAboutMe(mockUserAllFields, document);
+  });
+
+  it('is populated with the correct image.', () => {
+    const userImage = document.getElementById(dashboard.USER_IMAGE_ID);
+    expect(userImage.src).toBe(mockUserAllFields.image);
+  });
+
+  it('shows the right email button', () => {
+    const userEmailButton = document.getElementById(dashboard.USER_EMAIL_ID);
+    expect(userEmailButton).toBeVisible();
+    expect(userEmailButton.href).toBe('mailto:' + mockUserAllFields.email);
+  });
+
+  it('has the right GitHub URL', () => {
+    const userGithubButton = document.getElementById(dashboard.USER_GITHUB_ID);
+    expect(userGithubButton.href).toBe(mockUserAllFields.gitHubURL);
+  });
+
+  it('is populated with the correct location', () => {
+    expect(screen.getByText(mockUserAllFields.location))
+        .not.toBeNull();
+  });
+
+  it('is populated with the correct name', () => {
+    expect(screen.getByText(mockUserAllFields.name)).not.toBeNull();
+  });
+
+  it('is populated with the correct bio', () => {
+    expect(screen.getByText(mockUserAllFields.bio)).not.toBeNull();
+  });
+
+  it('is populated with the correct interest tags', () => {
+    mockUserAllFields.interestTags.forEach(function(interest) {
+      expect(screen.getByText(interest)).not.toBeNull();
+    });
+  });
+
+  it('is populated with the correct project tags', () => {
+    mockUserAllFields.projects.forEach(function(project) {
+      const projectTag = screen.getByText(project.name);
+      expect(projectTag).not.toBeNull();
+      expect(projectTag.href)
+          .toEqual(
+              dashboard.createProjectBreakoutURL(project.repositoryId)
+                  .toString());
+    });
+  });
+
+  it('does not display undefined', () => {
+    expect(screen.queryByText('undefined')).toBeFalsy();
+  });
+});
+
+describe('About me card of a user with missing optional fields', () => {
+  beforeAll(() => {
+    resetDocumentBody();
+    dashboard.createAboutMe(mockUserMissingOptionalFields, document);
+  });
+
+  it('hides the email button', () => {
```

Comment:
Q: Once again, I am just wondering if it is ever possible to successfully enter the dashboard page and retrieve user info without that user having input an email.